conseil-schema/src/main/scala/tech/cryptonomic/conseil/schema/GenSchema.scala

```diff
@@ -1,51 +1,147 @@
 package tech.cryptonomic.conseil.schema
 
-import pureconfig.{loadConfig, CamelCase, ConfigFieldMapping}
-import pureconfig.generic.auto._
-import pureconfig.generic.ProductHint
+import java.net.URI
+import java.nio.file.Paths
+
+import cats.implicits._
+import slick.codegen.SourceCodeGenerator
+import slick.basic.DatabaseConfig
+import slick.jdbc.PostgresProfile
+import slick.model.{Model, Table}
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import cats.effect._
 
 /**
   * Uses Slick's code-generation capabilities to infer code from Conseil database schema.
   * See http://slick.lightbend.com/doc/3.2.1/code-generation.html
   */
 object GenSchema extends App {
 
-  sealed trait DatabaseConfig {
-    def databaseName: String
-    def user: String
-    def password: String
-  }
+  /* configuration section needed, both locally or in the classpath-provided file,
+   * where the db connection definition is expected
+   * It uses a URI fragment syntax, following the rules of the Slick Generator
+   */
+  val configSection = "#slickgen"
+
+  val confUri = {
+    val externallyProvided = args.headOption.map(
+      confFile => Paths.get(confFile).toUri.resolve(configSection)
+    )
+    val classpathProvided = new URI(configSection) //this looks up in the standard application.conf as a resource
+    println(s"""
+    | Loading schema definitions from db configurations located at
+    | - externally provided: $externallyProvided
+    | - classpath fallback: application.conf$classpathProvided
+    |
+    | The source files will be generated under the folder /tmp/conseil-slick
```

Comment:
I guess under the folder: `slickgen.codegen.outputDir`, meaning configuration file (that might create a bit confusion, that this path is hardcoded somewhere and I hope it is not :D )